ladybug/datatype.py

```diff
@@ -1,832 +1,1970 @@
-        standard: 'SI' or 'IP' (Default: 'SI')
-        nickname: Optional nickname for data (e.g. Dew Point Temperature)
-    """
 
-    __slots__ = ()
-    maximum = 40
+class DryBulbTemperature(Temperature):
+    name = 'Dry Bulb Temperature'
+    abbreviation = 'DBT'
+    min_epw = -70
+    max_epw = 70
+    missing_epw = 99.9
 
 
-class Time(DataPoint):
-    """Time.
+class DewPointTemperature(Temperature):
+    name = 'Dew Point Temperature'
+    abbreviation = 'DPT'
+    min_epw = -70
+    max_epw = 70
+    missing_epw = 99.9
 
-    Attributes:
-        value: Input value
-        datetime: Date time data for this value (Default: None)
-        standard: 'SI' or 'IP' (Default: 'SI')
-        nickname: Optional nickname for data (e.g. Dew Point Temperature)
-    """
 
-    __slots__ = ()
-    minimum = 0
-    missing = 99 * 3600
-    value_type = int
-    unitSI = 'second'
-    unitIP = 'second'
+class SkyTemperature(Temperature):
+    name = 'Sky Temperature'
+    abbreviation = 'Tsky'
 
-    def __init__(self, value, datetime=None, standard='SI', nickname=None):
-        """Init class."""
-        DataPoint.__init__(self, value, datetime, standard, nickname)
 
+class AirTemperature(Temperature):
+    name = 'Air Temperature'
+    abbreviation = 'Tair'
 
-class Tenth(DataPoint):
-    """Tenth.
 
-    Attributes:
-        value: Input value
-        datetime: Date time data for this value (Default: None)
-        standard: 'SI' or 'IP' (Default: 'SI')
-        nickname: Optional nickname for data (e.g. Dew Point Temperature)
-    """
+class RadiantTemperature(Temperature):
+    name = 'Radiant Temperature'
+    abbreviation = 'Trad'
 
-    __slots__ = ()
-    minimum = 0
-    maximum = 10
-    missing = 99
-    value_type = int
-    unitSI = None
-    unitIP = None
 
-    def __init__(self, value, datetime=None, standard='SI', nickname=None):
-        """Init class."""
-        DataPoint.__init__(self, value, datetime, standard, nickname)
+class OperativeTemperature(Temperature):
+    name = 'Operative Temperature'
+    abbreviation = 'To'
 
 
-class Thousandths(DataPoint):
-    """Thousandths.
+class MeanRadiantTemperature(Temperature):
+    name = 'Mean Radiant Temperature'
+    abbreviation = 'MRT'
 
-    Attributes:
-        value: Input value
-        datetime: Date time data for this value (Default: None)
-        standard: 'SI' or 'IP' (Default: 'SI')
-        nickname: Optional nickname for data (e.g. Dew Point Temperature)
-    """
 
-    __slots__ = ()
-    minimum = 0
-    missing = 999
-    value_type = float
-    unitSI = 'thousandths'
-    unitIP = 'thousandths'
+class StandardEffectiveTemperature(Temperature):
+    name = 'Standard Effective Temperature'
+    abbreviation = 'SET'
 
-    def __init__(self, value, datetime=None, standard='SI', nickname=None):
-        """Init class."""
-        DataPoint.__init__(self, value, datetime, standard, nickname)
 
+class UniversalThermalClimateIndex(Temperat
```

Comment:
@mostaphaRoudsari , I finished breaking this down into separate .py files by data type as we had discussed.  Having all of the types in their own folder definitely feels right to me and I am glad that you recommended it.  The one thing I could really use feedback on now is about how the rest of the plugin interfaces with these types.  Specifically, I still have the rest of the library interfacing with the types folder by means of a datatype.py file that sits at a level above the types folder (at the root of the ladybug library).  This datatype.py file basically handles the lookup of data types by their name or the lookup of data types by unit, which is how I imagine most users and UIs interfacing with the types namespace.  Having all of the interaction handled at one point also seems cleaner to me since it allows us to easily pull out the types folder if we ever need to and then all that we have to do is put some code to patch things up in the datatype.py file. The question I have been asking myself is whether the root of the ladybug library is the right place for this file datatype.py or should it be sitting within the types folder?  I am also asking myself if it should retain the name datatype.py.

Reply:
> This datatype.py file basically handles the lookup of data types by their name or the lookup of data types by unit, which is how I imagine most users and UIs interfacing with the types namespace.

This is most likely what you want to do inside the `__init__.py` file. Let me check the files and I will leave a comment there.

Reply:
As I said elsewhere, all of the magic is not called from __init__.py